presto-main/src/main/java/com/facebook/presto/operator/scalar/annotations/ParametricScalarImplementation.java

```diff
@@ -569,7 +595,7 @@ private void inferSpecialization(Method method, Class<?> parameterType, String t
                 if (annotation instanceof TypeParameter) {
                     checkTypeParameters(parseTypeSignature(((TypeParameter) annotation).value()), typeParameterNames, method);
                 }
-                constructorDependencies.add(createDependency(annotation, literalParameters));
+                constructorDependencies.add(createDependency(annotation, literalParameters, null));
```

Comment:
This is not correct. You can not blindly pass `null`. It should be treated the same way as the one in `parseArguments`.

However, once you move the logic for parsing invocation convention into createDependency, this should not be a problem any more.